src/prymitive2.ts

```diff
@@ -0,0 +1,210 @@
+    _levelName: string;
+  
+    constructor(directionName: string, levelName: string, area: string, status: ('active' | 'pending'), students: Student[]) {
+      this._directionName = directionName;
+      this._levelName = levelName;
+      this._area = area;
+      this._status = status;
+      this._students = students;
+    }
+
+    set status (groupStatus:('active' | 'pending')) {
+        this._status = groupStatus
+    }
+    get area () : string {
+        return this._area
+    }
+    get status () : string {
+        return this._status
+    }    
+    get studentst () : Student[] {
+        return this._students
+    }    
+    get directionName () : string {
+        return this._directionName
+    }    
+    get levelName () : string {
+        return this._levelName
+    }
+    addStudent(student: Student) : void {
+        this._students.push(student)
+    }
+    removeGroup (studentToRemove: Student) : void {
+        this._students = this._students.filter(item => item !== studentToRemove );
+    }    
+
+    showPerformance() :Student [] {
+      const sortedStudents = this._students.toSorted((a, b) => b.getPerformanceRating() - a.getPerformanceRating());
+      return sortedStudents;
+    }
+}
+  
+type grade = {[workName:string]:number}; //it can be not only 1 key-value up here, how to prevent that???
+type visit = {[lesson: string]:boolean};//it can be not only 1 key-value up here, how to prevent that???
+
+class Student {
+    // implement 'set grade' and 'set visit' methods
+    _firstName: string;
+    _lastName: string;
+    _birthYear: number;
+    _grades: grade[] = []; 
+    _visits: visit[] = []; 
+  
+    constructor(firstName: string, lastName: string, birthYear: number, grades:grade[], visits:visit[]) {
+      this._firstName = firstName;
+      this._lastName = lastName;
+      this._birthYear = birthYear;
+      this._grades = grades;
+      this._visits= visits;
+    }
+  
+    get fullName(): string {
+      return `${this._lastName} ${this._firstName}`;
+    }
+  
+    set fullName(value: string) {
+      [this._lastName, this._firstName] = value.split(' ');
+    }
+    
+    set grade (grage:grade){
+        const workName : string = Object.keys(grage)[0]
+        const index :number = this._grades.findIndex(el => Object.keys(el)[0] === workName)
+        if (index >=0) {
+            this._grades[index][workName] = grage[workName]
+        }
+        else{
+            this._grades.push(grage)
+        }
+    }
+
+    set visits (visitToSet: visit){
+        const lesson: string = Object.keys(visitToSet)[0];
+        const index: number = this._visits.findIndex( el => Object.keys(el)[0] === lesson)
+        if (index >=0) {
+            this._visits[index][lesson] = visitToSet[lesson]
+        }
+        else{
+            this._visits.push(visitToSet)
+        }
+    }
+
+    get age():number {
+      return new Date().getFullYear() - this._birthYear;
+    }
+  
+    getPerformanceRating() {
```

Comment:
Please remember to include type annotations for the return values of your functions.